### PR TITLE
[Tests-Only] [full-ci] enforce strict type checking on acceptance test code

### PR DESCRIPTION
**Pull request by dpakach**
## Description

enforce strict type checking on acceptance test code

## Related Issue

- Related https://github.com/owncloud/QA/issues/694

## Motivation and Context

to make acceptance tests code more reliable by strict type checking.

## How Has This Been Tested?

:robot:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliTrashbin-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/33657/118/1

**Comment by phil-davis** [comment]
@dpakach I added full-ci to the PR title - you probably want to see all the acceptance test failures.

File: tests/TestHelpers/LoggingHelper.php [hunk 1]
```diff
@@ -99,7 +99,7 @@ public static function getLogFileContent(
 		?string $adminPassword,
 		?string $xRequestId = '',
 		?int $noOfLinesToRead = 0
-	):SimpleXMLElement {
+	):array {
```
Comment (phil-davis) [inline tests/TestHelpers/LoggingHelper.php:102]: PHPdoc is now out-of-date - line 93 `@return SimpleXMLElement` also needs changing.

(some time we should decide if we still want/need that sort of PHPdoc any more)

File: tests/acceptance/features/bootstrap/AppConfigurationContext.php [hunk 2]
```diff
@@ -253,7 +253,7 @@ public function getParameterValueFromXml(
 	 * @return boolean
 	 */
 	public function parameterValueExistsInXml(
-		string $xml,
+		SimpleXMLElement $xml,
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/AppConfigurationContext.php:256]: PHPdoc also needs updating.

File: tests/acceptance/features/bootstrap/FederationContext.php [hunk 3]
```diff
@@ -112,7 +112,7 @@ public function userFromServerSharesWithUserFromServerUsingTheSharingAPIWithPerm
 		$this->featureContext->createShare(
 			$sharerUser,
 			$sharerPath,
-			6,
+			(string) 6,
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/FederationContext.php:115]: Looks a bit odd, but I suppose these share types are "numbers" that just get moved around in requests and responses in ASCII numeric-string form.

In other places you put some of these numbers in quotes.

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 4]
```diff
@@ -995,7 +995,7 @@ public function createShare(
 		?string $path = null,
 		?string $shareType = null,
 		?string $shareWith = null,
-		?string $publicUpload = null,
+		?bool $publicUpload = null,
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Sharing.php:998]: PHPdoc also needs updating.

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 5]
```diff
@@ -4286,13 +4286,14 @@ public function userDeletesEverythingInFolder(
...
-			1
+			'1'
 		);
 		$elementList = $responseXmlObject->xpath("//d:response/d:href");
 		if (\is_array($elementList) && \count($elementList)) {
 			\array_shift($elementList); //don't delete the folder itself
 			$davPrefix = "/" . $this->getFullDavFilesPath($user);
 			foreach ($elementList as $element) {
+				$element = (string)$element;
 				$element = \substr($element, \strlen($davPrefix));
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebDav.php:4297]:
```suggestion
				$element = \substr((string)$element, \strlen($davPrefix));
```

does this work?

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 6]
```diff
@@ -4940,8 +4941,11 @@ public function findEntryFromPropfindResponse(
 		?string $user = null,
 		string $type = "files"
 	) {
+		$trimmedEntryNameToSearch = '';
 		// trim any leading "/" passed by the caller, we can just match the "raw" name
-		$trimmedEntryNameToSearch = \trim($entryNameToSearch, "/");
+		if ($entryNameToSearch) {
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebDav.php:4946]:
```suggestion
		if ($entryNameToSearch !== null) {
```

I am scared that `$entryNameToSearch` will be "0" some day and evaluate to `false`

**Review by phil-davis — APPROVED** [review]

**Pull request merged**